packages/react/src/drafts/TabPanels/TabPanels.tsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react'
+import styled from 'styled-components'
+import {get} from '../../constants'
+import {TabContainerElement} from '@github/tab-container-element'
+import {createComponent} from '../../utils/custom-element'
+import sx, {type SxProp} from '../../sx'
+import type {ComponentProps} from '../../utils/types'
+import getGlobalFocusStyles from '../../internal/utils/getGlobalFocusStyles'
+
+const TAB_CLASS = 'TabPanel-tab'
+
+const TabContainer = styled(createComponent(TabContainerElement, 'tab-container'))`
+  & > :not([role='tabpanel']) {
+    display: inline-block;
+  }
+
+  &::part(tablist-wrapper) {
+    margin-top: 0;
+    margin-bottom: 16px;
+    border-bottom: 1px solid ${get('colors.border.default')};
+  }
+
+  &:not(:defined) [role='tabpanel'] {
+    margin-top: 17px;
+    display: none;
+  }
+
+  &:not(:defined) [role='tab']:nth-of-type(1)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(1),
+  &:not(:defined) [role='tab']:nth-of-type(2)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(2),
+  &:not(:defined) [role='tab']:nth-of-type(3)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(3),
+  &:not(:defined) [role='tab']:nth-of-type(4)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(4),
+  &:not(:defined) [role='tab']:nth-of-type(5)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(5),
+  &:not(:defined) [role='tab']:nth-of-type(6)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(6),
+  &:not(:defined) [role='tab']:nth-of-type(7)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(7),
+  &:not(:defined) [role='tab']:nth-of-type(8)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(8),
+  &:not(:defined) [role='tab']:nth-of-type(9)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(9),
+  &:not(:defined) [role='tab']:nth-of-type(10)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(10) {
+    display: block;
+  }
+
+  &:not(:defined):not(:has([aria-selected='true'])) [role='tabpanel']:first-of-type {
+    display: block;
+  }
+
+  &:not(:has([aria-selected='true'])) [role='tab'] ~ [role='tab'] {
+    color: ${get('colors.fg.muted')};
+    background-color: transparent;
+    border: 1px solid transparent;
+  }
+
+  &:not(:has([aria-selected='true'])) [role='tab'],
+  & [role='tab'][aria-selected='true'] {
+    color: ${get('colors.fg.default')};
+    border-color: ${get('colors.border.default')};
+    border-top-right-radius: ${get('radii.2')};
+    border-top-left-radius: ${get('radii.2')};
+    background-color: ${get('colors.canvas.default')};
+  }
+
+  &:not(:defined):not(:has([aria-selected='true'])) [role='tab'] ~ [role='tab'],
+  &:not(:defined):has([aria-selected='true']) [role='tab']:not([aria-selected='true']) {
+    padding: 8px 16px;
+    border-bottom: 1px solid ${get('colors.border.default')};
+  }
+
+  &:not(:defined) :not([role='tabpanel']) {
+    vertical-align: top;
+  }
+
+  ${sx};
+`
+
+export type TabPanelsProps = ComponentProps<typeof TabContainer>
```

Comment:
I'm having a hard time getting these types to suggest anything, wanted to ask is this specific to my setup or are others seeing this as well? I think a specific case is that `onTabContainerChange` is not showing up and, if defined, the `event` type is `any`.

Reply:
I think maybe I need to redefine the events using the createContainer `Events` stuff 🤔